crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2085,6 +2117,21 @@ impl<'db> InstanceType<'db> {
     pub fn is_known(&self, known_instance: KnownInstance) -> bool {
         self.known == Some(known_instance)
     }
+
+    pub fn to_class_literal_type(self) -> ClassLiteralType<'db> {
```

Comment:
Minor: the name `to_…` might suggest that instance types are more or less equivalent to class literal types, since they can be converted so easily(?). Maybe something like
```suggestion
    pub fn corresponding_class_literal_type(self) -> ClassLiteralType<'db> {
```

Reply:
I was very unsure whether this method and `ClassLiteralType::to_instance_type` were more helpful than they were confusing. Your comment here convinces me that they're on the "more confusing" side of the line 😆

I removed them in https://github.com/astral-sh/ruff/pull/14108/commits/22d7a3b0802f31f591fff52e31f7b9f44a3b2841, in favour of a `Type::anonymous_instance` helper function.